test/integration/targets/dnf/tasks/dnf.yml

```diff
@@ -504,6 +504,7 @@
     that:
         - "dnf_result is failed"
         - "not dnf_result.changed"
+        - "'Failure downloading' in dnf_result.msg or 'Failed to download' in dnf_result.msg"
```

Comment:
The second condition is for the dnf5 module, which already handles the error properly (and does not call fetch_file at all).